### PR TITLE
remove ~= from dependency definitions

### LINKED ISSUE #88: Standardize Dependency Definitions
### Housekeeping

- [X] I am a maintainer of dbt-common

### Short description

As OSS at the org we are standardizing on using >=,< instead of ~= for dependency definitions.  Update dbt-common dependencies to follow this pattern.

### Acceptance criteria

- No uses of ~=, instead use >=,<

### Suggested Tests

None

### Impact to Other Teams

There should be none.  The definitions should be equivalent.

### Will backports be required?

No

### Context

`~=` issues; `pip` cannot resolve:

- Different "grains": `dbt-adapters` depends on `mashumaro~=3.9` and `dbt-common` depends on `mashumaro~=3.9.0`
- Different notations: `dbt-adapters` depends on `mashumaro<4.0` and `dbt-common` depends on `mashumaro~=3.9`

### PR DESCRIPTION
**Pull request by emmyoop**
resolves #88 

### Description

Replace `~=` with `>=,<` in dependency definitions.

Note: This does not address #14 

### Checklist

- [ ] I have read [the contributing guide](https://github.com/dbt-labs/dbt-common/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [ ] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [ ] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-common/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -27,39 +27,39 @@ classifiers = [
   "Programming Language :: Python :: Implementation :: PyPy",
 ]
 dependencies = [
-  "agate~=1.7.0",
+  "agate>=1.7.0,<1.8",
   "colorama>=0.3.9,<0.5",
   "isodate>=0.6,<0.7",
-  "jsonschema~=4.0",
+  "jsonschema>=4.0,<5.0",
   "Jinja2~=3.0",
```
Comment (emmyoop) [inline pyproject.toml:34]: Didn't update this to prevent conflicts with #85

**Review by aranke — APPROVED** [review]
lgtm, ty!

File: pyproject.toml [hunk 2]
```diff
@@ -27,39 +27,39 @@ classifiers = [
...
-    "types-Jinja2~=2.11",
-    "types-jsonschema~=4.17",
-    "types-protobuf~=4.24.0",
-    "types-python-dateutil~=2.8",
-    "types-PyYAML~=6.0",
+    "mypy>=1.3,<2.0",
+    "pytest>=7.3,<8.0",  # needed for linting tests
+    "types-Jinja2>=2.11,<3.0",
+    "types-jsonschema>=4.17,<5.0",
+    "types-protobuf>=4.24,<5.0",
```
Comment (mikealfare) [inline pyproject.toml:53]: `>=4.24,<4.25` However, this is very specific. The decision to open this up to the major might be intentional and safe.

**Review by mikealfare — APPROVED** [review]

File: pyproject.toml (hunk 2 above)
Comment (emmyoop) [reply pyproject.toml:53]: I opened it up to match protobuf itself.  It didn't start until 4.24 which is why the lower bounds don't match.

**Pull request merged**